### ISSUE TITLE
Feature: countdown zur Wahl

### ISSUE THREAD

**Issue by b90g** [verdigado/sunflower#787]
TYPO3 hats, WordPress noch nicht.

**Comment by FloJuh** [comment]
Christian hatte 2021 angeboten uns sowas zu programmieren: https://chatbegruenung.de/channel/sunflower-wordpress?msg=bx78Yg2QavAcNJDRF

**Comment by levino** [comment]
Kann man das nicht einfach mit ein bisschen html und js als unabhängiges Element einbetten?

**Comment by albig** [comment]
Klar. Teil gern deine Lösung.

**Comment by levino** [comment]
Courtesy of chat gpt:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Countdown Timer</title>
  <script src="https://cdn.tailwindcss.com"></script>
</head>
<body class="flex items-center justify-center h-screen bg-gray-900 text-white">
  <div class="text-center space-y-4">
    <h1 class="text-4xl font-bold">Countdown to January 23, 2025</h1>
    <div id="countdown" class="flex justify-center space-x-6 text-2xl">
      <div class="flex flex-col items-center">
        <span id="days" class="font-mono text-5xl transition-all">0</span>
        <span>Days</span>
      </div>
      <div class="flex flex-col items-center">
        <span id="hours" class="font-mono text-5xl transition-all">0</span>
        <span>Hours</span>
      </div>
      <div class="flex flex-col items-center">
        <span id="minutes" class="font-mono text-5xl transition-all">0</span>
        <span>Minutes</span>
      </div>
      <div class="flex flex-col items-center">
        <span id="seconds" class="font-mono text-5xl transition-all">0</span>
        <span>Seconds</span>
      </div>
    </div>
  </div>

  <script>
    const countdown = () => {
      const targetDate = new Date("2025-01-23T00:00:00+01:00"); // Berlin time
      const now = new Date();
      const difference = targetDate - now;

      if (difference <= 0) {
        document.getElementById("countdown").innerHTML = "<div class='text-3xl font-bold'>The date has arrived!</div>";
        return;
      }

      const days = Math.floor(difference / (1000 * 60 * 60 * 24));
      const hours = Math.floor((difference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
      const minutes = Math.floor((difference % (1000 * 60 * 60)) / (1000 * 60));
      const seconds = Math.floor((difference % (1000 * 60)) / 1000);

      document.getElementById("days").textContent = days;
      document.getElementById("hours").textContent = hours;
      document.getElementById("minutes").textContent = minutes;
      document.getElementById("seconds").textContent = seconds;
    };

    countdown();
    setInterval(countdown, 1000);
  </script>
</body>
</html>
```

**Comment by alexschnapper** [comment]
@albig und @levino finde das cool. Habe mir mal kleine Gedanken gemacht und die auch in meiner Testinstanz getestet. Und als Repo in github hochgeladen. => https://github.com/alexschnapper/sunflower_countdown 
Wollt ihr mal da drüber schauen? Soll ich das mal entspechend hier im repo forken und dann ggfs ein Pull Request branch name issue 787 erstellen?
😄

**Comment by albig** [comment]
Danke für deinen Aufschlag @alexschnapper . Ich hab's ausprobiert und es funktioniert grundsätzlich. Aber "nur" als Widget, was beim Sunflower-Theme ja nur im Footer ist.

Ich hab naiv bei den Blöcken mal "Counter" eingegeben und mir hat WordPress dann https://wordpress.org/plugins/countdown-time vorgeschlagen. Das kann man man sowohl als Gutenberg-Block als auch als Widget verwenden.

![Image](https://github.com/user-attachments/assets/fb825bb4-ba63-40dd-8eb1-123f37386744)

Vielleicht sollten wir zunächst klären, wie der Counter aussehen und wo er platziert werden soll. Ich könnte mir den ja auch "frei" irgendwo oben vorstellen und die Einstellungen über die Sunflower-Einstellungen.

**Comment by FloJuh** [comment]
Ein Gutenberg-Block wäre praktischer. Ich würde die Einstellungen jedoch direkt im Block vornehmen und nicht in den Sunflower-Einstellungen, da sie dort möglicherweise schwerer zu finden sind und dies zu nachfragen führen könnte. Auch könnte man so verschiedene Countdowns erstellen

**Comment by albig** [comment]
Für ein Block sollten die Einstellungen auch dort passieren. Ein losgelöster Counter z.B. im Header wäre dann was für die Sunflower-Einstellungen.

**Comment by levino** [comment]
Also ich denke, es sollte erst einmal klar gesagt werden, ob so ein Block / Widget tatsächlich als Feature aufgenommen werden soll, damit sich @alexschnapper nicht zu viel Arbeit macht. Ich bin ja quasi nur ein Spaziergänger, der sich hier über den Gartenzaun lehnt und ungefragt Eure Arbeit am Fundament der neuen Gartenhütte kommentiert. Aber ich halte das für ein Feature, was fast niemand braucht und auch keinen value zu einer Seite addet: Wieso sollte der Ortsverband Hintertupfingen irgendwelchen Leuten zeigen wollen, wie lange es noch bis zur Wahl ist? Und wenn die das wollen, dann können sie einfach das HTML oben in die Seite werfen und haben es erledigt. Solange kein klarer Konsens besteht, dass das Feature rein soll, sorgen irgendwelche "Vorschläge" höchstens dafür, dass @alexschnapper sich Mühe macht, ohne die Chance zu haben, dass das irgendwann gemerged wird. Sehr frustrierende Erfahrung. Insofern schlage ich vor, dass dazu erst einmal eine Aussage (mit Specs!) von jemandem kommt, der nachher auch willig ist, das zu mergen.

**Comment by alexschnapper** [comment]
Danke @levino - ich hätte auch kein Problem, wenn gesagt wird, dass es eher nicht zur Frage steht.
Ich war ja auch eher jmd der einfach ungefragt über den Gartenzaun einfach mal so was dazu beigetragen hat, ohne vorher zu connecten ;-) 
Ich bin aber auch  völlig d'accord damit,  dass es etwas wäre, was nur sehr wenige nutzen würden...

**Comment by levino** [comment]
Ja, mir geht es vor allem darum, dass erst einmal klar ist, ob das Feature aufgenommen wird, bevor hier über Specs gesprochen wird. Ich wollte nur Frust vorbeugen. Leider scheint hier auch niemand eine Entscheidung treffen zu wollen. Mein Instinkt, dass es hier Schwierigkeiten geben könnte, scheint mir insofern nicht ganz unbegründet.
